### DISCUSSION TITLE
Getting dlrs error when trying to create a child case using survey data mapping

### DISCUSSION THREAD

**Discussion by raghavic1995** [SFDO-Community/declarative-lookup-rollup-summaries#1338] [Q&A]
Hi,

We have built a survey in which child case gets created through data mapping whenever the customer provides a negative feedback. Currently, case is not getting created and we are getting an error stating that insufficient or read only access.

Error:

Error element rc_966063d0_0d9e_4be0_a7c5_f20da96d9568 (FlowRecordCreate).
This error occurred when the flow tried to create records: CANNOT_INSERT_UPDATE_ACTIVATE_ENTITY: dlrs_CaseTrigger: execution of AfterInsert caused by: System.DmlException: Update failed. First exception on row 1 with id 00558000001T5NHAA0; first error: INSUFFICIENT_ACCESS_OR_READONLY, Cannot insert a user with this profile: [] (dlrs) Trigger.dlrs_CaseTrigger: line 7, column 1. You can look up ExceptionCode values in the SOAP API Developer Guide.

Flow Details
Flow API Name: dm_ee16db12_63aa_4808_aef4_f1ab53f7cdb8
Type: Survey Enrich Flow
Version: 4
Status: Active
Org: Imerys (00D1q0000004nvn)
Flow Interview Details
Interview Label: dm_ee16db12_63aa_4808_aef4_f1ab53f7cdb8
Interview GUID: 8811e6ef47b50560266510c621f318793bf54c5-5e2e
Current User: Automated Process (00558000001T5NH)
Start time: 18/04/2023 11:44
Duration: 3 seconds
How the Interview Started
Automated Process (00558000001T5NH) started the flow interview.
API Version for Running the Flow: 57
Some of this flow's variables were set when the interview started.
q_6cc1fa0c_fe62_44e0_a2f2_0a58e46742e9 = 2
invitationId = 0Ki1q0000001Aaf
GET RECORDS: surveySubjectRecordLookup_Case
Find one SurveySubject record where:
SurveyInvitationId Equals {!invitationId} (0Ki1q0000001Aaf)
AND SubjectEntityType Equals Case
Result
Successfully found record.
{!surveySubjectLookupSubjectIdVar} = 5001q00000LTKYLAA5

GET RECORDS: targetRecordLookup_Case
Find all Case records where:
Id Equals {!surveySubjectLookupSubjectIdVar} (5001q00000LTKYLAA5)
Store those records in {!Case}.
Save these field values in the variable: AccountId, Account__c, ContactId, CustomerService2__c, Id
Result
Successfully found records.
GET RECORDS: surveyParticipantRecordLookup_ContactDetails
Find one SurveyInvitation record where:
Id Equals {!invitationId} (0Ki1q0000001Aaf)
Result
Successfully found record.
{!surveyParticipantLookupIdVar} = null

GET RECORDS: targetParticipantLookup_ContactDetails
Find all Contact records where:
Id Equals {!surveyParticipantLookupIdVar} (null)
Store those records in {!ContactDetails}.
Save these field values in the variable: Id
Result
Failed to find records.
DECISION: my_decision
Outcome executed: r_4eb7b516_0800_4c6b_b69d_bd49a5eb0b77
Outcome conditions:
1. {!q_6cc1fa0c_fe62_44e0_a2f2_0a58e46742e9} (2) Equals {!c_7ac35f7d_b231_4a31_b5a0_371af2deb624} (1)
2. {!q_6cc1fa0c_fe62_44e0_a2f2_0a58e46742e9} (2) Equals {!c_8d91d2be_f27c_44a0_a43f_a1aec24eaea5} (2)
3. {!q_6cc1fa0c_fe62_44e0_a2f2_0a58e46742e9} (2) Equals {!c_0279bfb1_7fb7_44eb_827e_b60ef84bcfb1} (3)
Logic: One condition must be true (OR)

CREATE RECORDS: Create child case
Create one Case record where:
AccountId = {!Case.AccountId} (0015800001JIKkjAAH)
Account__c = {!Case.Account__c} (Richard Baker Harrison - Ilford)
AutomaticAcknowledgmentMailtoContact__c = No
BusinessArea__c = HTS
BusinessHoursId = 01m58000000QBAz
ContactId = {!Case.ContactId} (0031q00000wsyuaAAA)
CustomerService2__c = {!Case.CustomerService2__c} (0051i000003E8QLAA0)
Description = Please follow up with the customer
ParentId = {!Case.Id} (5001q00000LTKYLAA5)
Priority = Medium
RecordTypeId = 0121q00000026KLAAY
Status = New
Subject = Negative Feedback
Tech_EmailCounter__c = true
Result
Failed to create record.

Expected result : child case gets created.
Actual result : getting error in the backend.

**Comment by aheber** [comment]
@raghavic1995 can you help me understand why you'd be seeing the error `INSUFFICIENT_ACCESS_OR_READONLY, Cannot insert a user with this profile: [] (dlrs) `
DLRS isn't likely causing this error but more likely happens to be the top of the stack when the error is surfaced. I'm not sure what logic would cause a User to be created, or what profile it is trying to create. Is that a function of the. Survey tool?
Could it be possible that you're rolling from the Case to the User and something about this execution context doesn't allow that update? Writing to users can have specific security behaviors that need to be managed correctly.

Is it possible that part of this is being interacted with by a Guest-type of user and that user doesn't have correct access to trigger these automations?

You may need to give that user the ability to perform some of these automations or move the rollups to Scheduled mode so that the rollup behavior can be done by a more privileged user. Noting that Guest Users would still need the ability to create Scheduled Item records for DLRS. (There should be a perm set for executing rollups, not configuring them)
